spec/rubocop/cop/style/redundant_array_flatten_spec.rb

```diff
@@ -0,0 +1,77 @@
+# frozen_string_literal: true
+
+RSpec.describe RuboCop::Cop::Style::RedundantArrayFlatten, :config do
+  it 'registers an offense for `x.flatten.join`' do
+    expect_offense(<<~RUBY)
+      x.flatten.join
+       ^^^^^^^^ Remove the redundant `flatten`.
+    RUBY
+
+    expect_correction(<<~RUBY)
+      x.join
+    RUBY
+  end
+
+  it 'registers an offense for `x.flatten.join(separator)`' do
+    expect_offense(<<~RUBY)
+      x.flatten.join(separator)
+       ^^^^^^^^ Remove the redundant `flatten`.
+    RUBY
+
+    expect_correction(<<~RUBY)
+      x.join(separator)
+    RUBY
+  end
+
+  it 'registers an offense for `x.flatten(depth).join`' do
+    expect_offense(<<~RUBY)
+      x.flatten(depth).join
+       ^^^^^^^^^^^^^^^ Remove the redundant `flatten`.
+    RUBY
+
+    expect_correction(<<~RUBY)
+      x.join
+    RUBY
+  end
+
+  it 'registers an offense for `x&.flatten&.join`' do
+    expect_offense(<<~RUBY)
+      x&.flatten&.join
+       ^^^^^^^^^ Remove the redundant `flatten`.
+    RUBY
+
+    expect_correction(<<~RUBY)
+      x&.join
+    RUBY
+  end
+
+  it 'does not register an offense for `x.flatten.foo`' do
+    expect_no_offenses(<<~RUBY)
+      x.flatten.foo
+    RUBY
+  end
+
+  it 'does not register an offense for `x.flatten`' do
+    expect_no_offenses(<<~RUBY)
+      x.flatten
+    RUBY
+  end
+
+  it 'does not register an offense for `flatten.join` without an explicit receiver' do
+    expect_no_offenses(<<~RUBY)
+      flatten.join
+    RUBY
+  end
```

Comment:
I excluded cases without an explicit receiver. If this were offending, cases such as this would be registered:
```ruby
class MyCustomClass
  def foo
    flatten.join
  end

  def flatten; end
  def join; end
end
```
which could be a false positive. On the other hand, it could also be a valid offense:
```ruby
class MyCustomArray < Array
  def flat_join(separator)
    flatten.join(separator)
  end
end
```

Let me know what makes more sense here 🙏